### DISCUSSION TITLE
boost-1.80.0.beta1

### DISCUSSION THREAD

**Discussion by userdocs** [arvidn/libtorrent#6964] [General]
Tried a build and had no obvious issues

https://github.com/boostorg/boost/releases/tag/boost-1.80.0.beta1

![image](https://user-images.githubusercontent.com/16525024/179321553-8d6987bd-2068-449d-8013-2ef4624ed2b0.png)

**Comment by xavier2k6** [comment]
>Tried a build and had no obvious issues

same (windows)

**Reply by userdocs** [reply]
Nice, I had one bug after I posted this with b2 (not cmake) where this failed

https://github.com/arvidn/libtorrent/blob/RC_2_0/Jamfile#L944

But it worked if I did this

```
local boost-build-version = "180" ;
```

Not sure if this a beta version bug and won't happen with non beta release. The versioning seems normal though in the `version.hpp`

```
#define BOOST_VERSION 108000

//
//  BOOST_LIB_VERSION must be defined to be the same as BOOST_VERSION
//  but as a *string* in the form "x_y[_z]" where x is the major version
//  number, y is the minor version number, and z is the patch level if not 0.
//  This is used by <config/auto_link.hpp> to select which library version to link to.

#define BOOST_LIB_VERSION "1_80"

#endif
```

I'll just wait and see when the full release comes out since there are no other obvious build concerns to worry about.

**Reply by xavier2k6** [reply]
Did you use the `b2` version from the beta (`4.9.2`)?

Take a look at the release notes:
https://github.com/bfgroup/b2/releases

**Reply by arvidn** [reply]
So, is version.boost-build missing in the new version?

**Reply by userdocs** [reply]
@arvidn I just see this

```
building boost from source directory:  /home/username/qbt-build/boost
Jamfile:943: in install-paths
ERROR: rule "version.boost-build" unknown in module "Jamfile</home/username/qbt-build/libtorrent>".
Jamfile:1062: in install-pkg-config
Jamfile:1093: in load-aux
/home/username/qbt-build/boost/tools/build/src/build/project.jam:378: in load-jamfile
/home/username/qbt-build/boost/tools/build/src/build/project.jam:64: in load
/home/username/qbt-build/boost/tools/build/src/build/project.jam:142: in project.find
/home/username/qbt-build/boost/tools/build/src/build-system.jam:618: in load
/home/username/qbt-build/boost/tools/build/src/kernel/modules.jam:294: in import
/home/username/qbt-build/boost/tools/build/src/kernel/bootstrap.jam:135: in module scope
```

**Reply by userdocs** [reply]
So is it this from [4.9.0](https://github.com/bfgroup/b2/releases/tag/4.9.0)

```
WARNING: This release removes the use of boost-build.jam and the
boost-build rule for initialization. The boost-build.jam is still
searched for and loaded to not break existing operations. But is considered
deprecated and will be removed in a future release.
```

**Reply by userdocs** [reply]
@arvidn so just messing around:

It works fine with 1.79.0 beta1

This printed (with 1.79)

```
local boost-build-version = [ SPLIT_BY_CHARACTERS [ version.boost-build ] : "-" ] ;
ECHO $(boost-build-version) ;
4.8 git
```

If i do `./b2 --version` with `1.80.0` i will get `4.9-git` but:

I found this file in the 1.80 downloaded source which appears to have everything required to process the boost build version

Major Minor Patch

https://raw.githubusercontent.com/boostorg/build/405d34a04d29519625c5edfe1f3bac3bc3dc3534/src/build/version.jam

So if I do this

```
local boost-build-version = [ modules.peek : JAM_VERSION ] ;
ECHO $(boost-build-version) ;
4 9 2
````

This seems like the right way to process the version number now?

I think this is the problem and part way to the fix?

**Reply by userdocs** [reply]
Just add i looked at older versions https://sourceforge.net/projects/boost/files/boost/1.69.0/

`boost_1_69_0/tools/build/src/build/version.jam`

```
import numbers ;

.major = "2018" ;
.minor = "02" ;
```

The format is the same?

**Reply by arvidn** [reply]
The version format changed at some point, I'm compensating for that already. ~However, I can't reproduce this issue using the boost-1.80 beta. are you using the git dev branch by any change?~ (I accidentally had brew's `b2` in path)

**Reply by arvidn** [reply]
https://github.com/arvidn/libtorrent/pull/6973
